### ISSUE TITLE
Ui Freeze when broker doesn't respond

### ISSUE THREAD

**Issue by wakatanka** [gpvigano/M2MqttUnity#2]
> * version of Unity3D 2018.all, 2019.1

Hi, i make mqtt autoconnect true and it works flawless, but if the mqtt broker is down all the ui remain freezed until timeout (put to 30sec). Is there a way to avoid this?

thanks for your great work

**Comment by gpvigano** [comment]
Hi, you can set the `timeoutOnConnection` field in `M2MqttUnityClient` to a smaller value, otherwise a best control over the connection should be implemented (in one of my projects I used a small timeout with a defined number of iterated trials, using a list of stored broker addresses and trying to connect to each of them until success).

I understand that this is more a workaround than a solution, sorry, maybe you can find a better support on [M2MQTT site](https://github.com/eclipse/paho.mqtt.m2mqtt) asking about `MqttClient`.

**Comment by takemehi** [comment]
The issue can be resolved by connecting in a seperate Thread instead of using a coroutine.

**Comment by gpvigano** [comment]
Right. That's the solution (maybe also connecting in a separate thread and waiting for the thread completion in the coroutine). Anyway attention must be paid working with threads (mostly in Unity), exceptions should be handled, etc. I'm not a skilled multi-threading programmer, a pull request would be welcome here...
